src/linker/Linker.Dataflow/ReflectionPatternContext.cs

```diff
@@ -99,7 +99,7 @@ public void RecordUnrecognizedPattern (int messageCode, string message)
 		public void Dispose ()
 		{
 #if DEBUG
-			if (_patternAnalysisAttempted && !_patternReported)
+			if (_patternAnalysisAttempted && !_patternReported && !_context.IsOptimizationEnabled (CodeOptimizations.ApplyTypeHierarchyAnnotation, (MemberReference) MemberWithRequirements))
```

Comment:
This is wrong - if there's a place where this throws we should fix that place and not relax this check. Reading the code I don't see a need for this (I don't see the place where this would throw).